atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/table/common/TableTasks.java

```diff
@@ -480,4 +479,40 @@ PartialDiffStats call(RangeRequest request, MutableRange range, DiffStrategy str
         private long rowsCopied = 0;
         private long cellsCopied = 0;
     }
+
+    private static class InterruptibleRangeExecutor {
+        private final ExecutorService exec;
+        private final int batchSize;
+        private final int threadCount;
+
+        InterruptibleRangeExecutor(ExecutorService exec, int batchSize, int threadCount) {
+            this.exec = exec;
+            this.batchSize = batchSize;
+            this.threadCount = threadCount;
+        }
+
+        void executeTask(String taskName, InterruptibleRangeTask task) throws InterruptedException {
+            BlockingWorkerPool pool = new BlockingWorkerPool(exec, threadCount);
+            for (final MutableRange range : getRanges(threadCount, batchSize)) {
+                if (Thread.currentThread().isInterrupted()) {
+                    log.info("Thread interrupted. Cancelling {} of range {}", taskName, range);
+                    break;
+                }
+                pool.submitTask(() -> {
```

Comment:
arguably `com.palantir.common.concurrent.BlockingWorkerPool#submitTask` should return `Future<Void>` so that if the submitting thread is interrupted, you can collect and cancel all of the submitted tasks